src/Symfony/Component/Messenger/Tests/EventListener/SendFailedMessageForRetryListenerTest.php

```diff
@@ -190,4 +192,49 @@ public function testEnvelopeKeepOnlyTheLast10Stamps()
 
         $listener->onMessageFailed($event);
     }
+
+    public function testRetriedEnvelopePassesToRetriedEvent()
+    {
+        $exception = new \Exception('no!');
+        $envelope = new Envelope(new \stdClass());
+
+        $sender = $this->createMock(SenderInterface::class);
+        $sender->expects($this->once())->method('send')->willReturnCallback(function (Envelope $envelope) {
+            return $envelope->with(new TransportMessageIdStamp(123));
+        });
+
+        $eventDispatcher = $this->createMock(EventDispatcherInterface::class);
+        $eventDispatcher->expects($this->once())->method('dispatch')->willReturnCallback(
+            function (WorkerMessageRetriedEvent $retriedEvent) {
+                $envelope = $retriedEvent->getEnvelope();
+
+                $transportIdStamp = $envelope->last(TransportMessageIdStamp::class);
+                $this->assertNotNull($transportIdStamp);
+
+                return $retriedEvent;
+            });
+
+        $senderLocator = $this->createMock(ContainerInterface::class);
```

Comment:
we'd better not mock the ContainerInterface, see #53535 for inspiration

Reply:
Thank you a lot for a review. 

Changed to `new Container()`. But is there some description somewhere why Symfony devs from now prefer to not mock container? Because in given pull request I see only changes, but no any info why it is bad practice from now.